osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/responses/from/core/processors/GetFirmwareFileResponseMessageProcessor.java

```diff
@@ -92,19 +93,26 @@ public void processMessage(final ObjectMessage message) throws JMSException {
           ResponseMessageResultType.NOT_OK,
           exception,
           this.responseMessageSender,
-          this.getFirmwareIdentification(message));
+          this.createUpdateFirmwareRequestDto(message));
     } finally {
       this.doConnectionPostProcessing(device, conn, messageMetadata);
     }
   }
 
-  private String getFirmwareIdentification(final ObjectMessage objectMessage) throws JMSException {
+  private UpdateFirmwareRequestDto createUpdateFirmwareRequestDto(final ObjectMessage objectMessage)
+      throws JMSException {
     final Serializable object = objectMessage.getObject();
 
     if (object instanceof ProtocolResponseMessage) {
-      final Serializable dataObject = ((ProtocolResponseMessage) object).getDataObject();
+
+      final ProtocolResponseMessage protocolResponseMessage = (ProtocolResponseMessage) object;
+      final Serializable dataObject = protocolResponseMessage.getDataObject();
+
       if (dataObject instanceof FirmwareFileDto) {
-        return ((FirmwareFileDto) dataObject).getFirmwareIdentification();
+
+        return new UpdateFirmwareRequestDto(
+            ((FirmwareFileDto) dataObject).getFirmwareIdentification(),
+            ((ProtocolResponseMessage) object).getDeviceIdentification());
```

Comment:
* Use the already declared protocolResponseMessage for obtaining the device identification.
* Consider creating a variable for fimwareFileDto.

The code may then become something like:
```suggestion

        FirmwareFileDto firmwareFileDto = (FirmwareFileDto) dataObject;
        return new UpdateFirmwareRequestDto(
            firmwareFileDto.getFirmwareIdentification(),
            protocolResponseMessage.getDeviceIdentification());
```